src/Provider/AbstractProvider.php

```diff
@@ -312,35 +341,53 @@ protected function getAccessTokenMethod()
         return 'POST';
     }
 
-    public function getAccessToken($grant = 'authorization_code', array $params = [])
+    /**
+     * Builds the access token URL's query string.
+     *
+     * @param array $params Query parameters
+     * @return string Query string
+     */
+    protected function getAccessTokenQuery(array $params)
+    {
+        return http_build_query($params);
+    }
+
+    /**
+     * Checks that a provided grant is valid, or attempts to produce one if the
+     * provided grant is a string.
+     *
+     * @param mixed $grant
+     * @return GrantInterface
+     */
+    protected function verifyGrant($grant)
     {
         if (is_string($grant)) {
-            $grant = $this->grantFactory->getGrant($grant);
-        } else {
-            $this->grantFactory->checkGrant($grant);
+            return $this->grantFactory->getGrant($grant);
         }
 
-        $defaultParams = [
-            'client_id'     => $this->clientId,
-            'client_secret' => $this->clientSecret,
-            'redirect_uri'  => $this->redirectUri,
-            'grant_type'    => (string) $grant,
-        ];
-
-        $requestParams = $grant->prepRequestParams($defaultParams, $params);
-        $requestParams = $this->httpBuildQuery($requestParams);
+        $this->grantFactory->checkGrant($grant);
+        return $grant;
+    }
 
+    /**
+     * Returns a prepared request for requesting an access token.
+     *
+     * @param array $params Query string parameters
+     */
+    protected function getAccessTokenRequest(array $params)
+    {
         $url = $this->urlAccessToken();
+        $query = $this->getAccessTokenQuery($params);
```

Comment:
You're right, I have fixed this in 4e0251110571ad7841d87dbe74f0f8d91383a32f. Wondering if there's a better way that we can test that as the tests pass either way.